File: source/windowsMobile/CommLayer/Messages/AckMessage.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CommLayerCompact.Messages
{
    /// <summary>
    /// Representa un mensaje de confirmación para los mensajes de tipo Safe
    /// Protocolo TCP
    /// </summary>
    internal class AckMessage : UnicastMessage 
    {
        /// <summary>
        /// El id del mensaje
        /// </summary>
        private Guid _messageId;

        /// <summary>
        /// Constructor
        /// </summary>
        public AckMessage() : base()
        {
            this.Type = MessageType.ACK;
            this.ProtocolType = MessageProtocolType.HLMP;
        }

        /// <summary>
        /// Constructor Parametrizado
        /// </summary>
        /// <param name="targetNetUser">El destinatario de este mensaje</param>
        /// <param name="messageId">El id del mensaje a confirmar</param>
        public AckMessage(NetUser targetNetUser, Guid messageId) : this()
        {
            this.TargetNetUser = targetNetUser;
            this.MessageId = messageId;
        }

        /// <summary>
        /// El id del mensaje a confirmar
        /// </summary>
        public Guid MessageId
        {
            get { return _messageId; }
            set { _messageId = value; }
        }

        /// <summary>
        /// Convierte las propiedades del mensaje en un paquete de bytes
        /// </summary>
        /// <returns>un paquete de bytes con las propiedades del mensaje</returns>
        public override byte[] makePack()
        {
            byte[] messageId = MessageId.ToByteArray(); //16 (0 - 15)
            return messageId;
        }

        /// <summary>
        /// Convierte un paquete de bytes en las propiedades del mensaje
        /// </summary>
        /// <param name="messagePack">El paquete de bytes</param>
        public override void unPack(byte[] messagePack)
        {
            MessageId = new Guid(messagePack);
        }

        /// <summary>
        /// Sobreescribe el metodo toString
        /// </summary>
        /// <returns>El string que representa este objeto</returns>
        public override string ToString()
        {
            return base.ToString() + "AckMessage: MessageId=" + MessageId.ToString();
        }

    }
}

File: source/java/hlmp/commlayer/messages/AckMessage.java
package hlmp.commlayer.messages;

import hlmp.commlayer.NetUser;
import hlmp.netlayer.Tools;
import hlmp.commlayer.constants.*;

import java.util.UUID;


public class AckMessage extends UnicastMessage {
	
	/**
	 * El id del mensaje a confirmar
	 */
    private UUID messageId;
    
    

	public AckMessage() {
		super();
		this.type = MessageType.ACK;
		this.protocolType = MessageProtocolType.HLMP;
	}

	public AckMessage(NetUser targetNetUser, UUID messageId) {
		this();
		this.targetNetUser = targetNetUser;
		this.messageId = messageId;
	}
	
	public UUID getMessageId() {
		return messageId;
	}

	public void setMessageId(UUID messageId) {
		this.messageId = messageId;
	}

	@Override
	public byte[] makePack() {
		byte[] messageId = Tools.UUIDtoBytes(this.messageId); //16 (0 - 15)
        return messageId;
	}

	@Override
	public void unPack(byte[] messagePack) {
		this.messageId = Tools.bytesToUUID(messagePack);
	}
	
	/**
	 * Sobreescribe el metodo toString
	 * @return El string que representa este objeto
	 */
	@Override
    public String toString(){
        return super.toString() + "AckMessage: MessageId=" + this.messageId.toString();
    }

}
